src/client/pages/Offer/ProductSelector/Selector.tsx

```diff
@@ -1,63 +1,88 @@
 import { useTextKeys } from 'utils/textKeys'
 import { MEDIA_QUERIES } from 'utils/mediaQueries'
+import { Product } from 'pages/Offer/types'
 import { MainProductCard } from './MainProductCard'
 import { AdditionalProductCard } from './AdditionalProductCard'
 
-type Product = {
-  id: string
-  name: string
-  description: string
-  price: string
-  image: string
-}
-
-type ProductSelectorProps = {
+type SelectorProps = {
   className?: string
-  mainProducts: Array<Product>
+  standaloneProducts: Array<Product>
   additionalProducts: Array<Product>
 }
 
-const getInitialState = ({
-  mainProducts,
+type State = {
+  standaloneProducts: Record<string, boolean>
+  additionalProducts: Record<string, boolean>
+}
+
+const getStateFromURL = ({
+  standaloneProducts,
   additionalProducts,
+  searchParams,
 }: {
-  mainProducts: Array<Product>
+  standaloneProducts: Array<Product>
   additionalProducts: Array<Product>
-}): {
-  mainProducts: Record<string, boolean>
-  additionalProducts: Record<string, boolean>
-} => {
-  const getProductIdByValueMap = (products: Array<Product>) =>
-    products.reduce(
-      (result, product) => ({
+  searchParams: string
+}): State => {
+  const selectedInsuranceTypes = new Set(
+    new URLSearchParams(searchParams).getAll('type'),
+  )
+
+  const getProductIdByValueMap = (quotes: Array<Product>) =>
+    quotes.reduce(
+      (result, quote) => ({
         ...result,
-        [product.id]: false,
+        [quote.id]: selectedInsuranceTypes.has(quote.id),
       }),
       {},
     )
 
   return {
-    mainProducts: getProductIdByValueMap(mainProducts),
+    standaloneProducts: getProductIdByValueMap(standaloneProducts),
     additionalProducts: getProductIdByValueMap(additionalProducts),
   }
 }
 
-export const ProductSelector = ({
+const getSearchParamsFromState = (state: State) => {
+  const selectedStandaloneProducts = Object.keys(
+    state.standaloneProducts,
+  ).filter((productId) => state.standaloneProducts[productId])
+  const selectedAdditionalProducts = Object.keys(
+    state.additionalProducts,
+  ).filter((productId) => state.additionalProducts[productId])
+  const selectedProducts = [
+    ...selectedStandaloneProducts,
+    ...selectedAdditionalProducts,
+  ]
+
+  const searchParams = new URLSearchParams()
+  selectedProducts.forEach((type) => searchParams.append('type', type))
+
+  return searchParams.toString()
+}
+
+export const Selector = ({
   className,
-  mainProducts,
+  standaloneProducts,
   additionalProducts,
-}: ProductSelectorProps) => {
+}: SelectorProps) => {
   const textKeys = useTextKeys()
+
+  const { search: searchParams } = useLocation()
+  const history = useHistory()
+
   const inputRef = useRef<HTMLInputElement | null>(null)
+
   const [selectedProductsByCategory, setSelectedProductsByCategory] = useState(
-    getInitialState({ mainProducts, additionalProducts }),
+    getStateFromURL({ standaloneProducts, additionalProducts, searchParams }),
   )
```

Comment:
can we get rid of this state and just rely on the URL instead?

would be good with one source of truth